frontend/src/robot/capabilities/MapControl.tsx

```diff
@@ -0,0 +1,111 @@
+            return (
+                <Typography color="error">
+                    Error loading persistent data state.
+                </Typography>
+            );
+        }
+    }
+
+    return (
+        <>
+            <FormGroup>
+                {!supported && (
+                    <Typography variant="body1" color="error">
+                        Toggling this feature is not supported by this robot.
+                    </Typography>
+                )}
+                <FormControlLabel control={
+                    <Switch disabled={!supported || persistentDataChanging}
+                        checked={supported ? persistentData?.enabled : true}
+                        onChange={(e) => {
+                            if (e.target.checked) {
+                                mutatePersistentData(true);
+                            } else {
+                                setDialogOpen(true);
+                            }
+                        }}/>
+                }
+                label="Enable persistent data"/>
+            </FormGroup>
+            <YesNoDialog title="Delete persistent data?" text="Do you really want to disable persistent data?"
+                open={dialogOpen} onClose={() => {
+                    setDialogOpen(false);
+                }} onAccept={() => {
+                    mutatePersistentData(false);
+                }}/>
+        </>
+    );
+};
+
+const MapResetControl: FunctionComponent = () => {
+    const [dialogOpen, setDialogOpen] = React.useState(false);
+    const {mutate: resetMap, isLoading: mapResetting} = useMapResetMutation();
+
+    return (
+        <>
+            <Typography variant="body1">
+                Delete persistent data
+            </Typography>
+            <LoadingButton loading={mapResetting} color="error" variant="outlined" onClick={() => {
+                setDialogOpen(true);
+            }}>Reset Map</LoadingButton>
+            <YesNoDialog title="Reset map?"
+                text="Do you really want to reset the map? This deletes the current map, all no-go zones and virtual walls."
+                open={dialogOpen} onClose={() => {
+                    setDialogOpen(false);
+                }} onAccept={() => {
+                    resetMap();
+                }}/>
+        </>
+    );
+};
+
+const MapControl: FunctionComponent = () => {
+    const [persistentMapControl, mapReset] = useCapabilitiesSupported(Capability.PersistentMapControl, Capability.MapReset);
+
+    if (!persistentMapControl && !mapReset) {
+        return null;
+    }
+
+    return (
+        <Card>
+            <CardContent>
+                <Typography variant="h6" gutterBottom>
+                    Persistent Data Configuration
+                </Typography>
+                <Divider/>
+                <Typography variant="body2">
+                    Persistent data is a feature of some robots which allows to save no-go areas and virtual walls. It
```

Comment:
This text might be slightly misleading as there are robots (e.g. Dreame vSLAM ones) where you can't disable persistent data. It's always enabled